src/common.h

```diff
@@ -45,59 +45,66 @@ enum RTR_FUZZ_TYPE {
+	PARAMETER_TYPE_MEM_BUFFER_ARRAY,	/* first size of buffers, then number of buffers, memory pointer at the end */
+	PARAMETER_TYPE_PRINTF_FORMAT,		/* printf format string followed by pointer to a va_list */
+	PARAMETER_TYPE_STRING_ARRAY,		/* fist the size of the array, then an array of char * */
+	PARAMETER_TYPE_IOVEC,				/* struct iovec */
+	PARAMETER_TYPE_UTSNAME,				/* struct utsname*  */
+	PARAMETER_TYPE_TIMEVAL,				/* struct timeval* */
+	PARAMETER_TYPE_TIMEZONE,			/* struct timezone* */
+	PARAMETER_TYPE_SSL,					/* OpenSSL's SSL* */
+	PARAMETER_TYPE_SSL_WITH_KEY,		/* OpenSSL's SSL* but attempt to dump the ssl key */
 #if HAVE_STRUCT_FLOCK
-#define PARAMETER_TYPE_STRUCT_FLOCK 25 /* fcntl's struct flock */
+	PARAMETER_TYPE_STRUCT_FLOCK,		/* fcntl's struct flock */
 #endif
 #if HAVE_STRUCT_FSTORE
-#define PARAMETER_TYPE_STRUCT_FSTORE 26 /* fcntl's struct fstore */
+	PARAMETER_TYPE_STRUCT_FSTORE,		/* fcntl's struct fstore */
 #endif
 #if HAVE_STRUCT_FPUNCHHOLE
-#define PARAMETER_TYPE_STRUCT_FPUNCHHOLE 27 /* fcntl's struct fpunchhole */
+	PARAMETER_TYPE_STRUCT_FPUNCHHOLE,	/* fcntl's struct fpunchhole */
 #endif
 #if HAVE_STRUCT_RADVISORY
-#define PARAMETER_TYPE_STRUCT_RADVISORY 28 /* fcntl's struct radvisory */
+	PARAMETER_TYPE_STRUCT_RADVISORY,	/* fcntl's struct radvisory */
 #endif
 #if HAVE_STRUCT_FBOOTSTRAPTRANSFER
-#define PARAMETER_TYPE_STRUCT_FBOOTSTRAPTRANSFER 29 /* fcntl's struct fbootstraptransfer */
+	PARAMETER_TYPE_STRUCT_FBOOTSTRAPTRANSFER,		/* fcntl's struct fbootstraptransfer */
 #endif
 #if HAVE_STRUCT_LOG2PHYS
-#define PARAMETER_TYPE_STRUCT_LOG2PHYS 30 /* fcntl's struct log2phys */
+	PARAMETER_TYPE_STRUCT_LOG2PHYS,		/* fcntl's struct log2phys */
 #endif
 #if HAVE_DECL_F_GETOWN_EX
-#define PARAMETER_TYPE_STRUCT_F_GETOWN_EX 31 /* fcntl's struct f_owner_ex */
+	PARAMETER_TYPE_STRUCT_F_GETOWN_EX,	/* fcntl's struct f_owner_ex */
 #endif
-#define PARAMETER_TYPE_PERM  32 /* write permission  */
-#define PARAMETER_TYPE_STRUCT_STAT 33 /* struct stat */
-#define PARAMETER_TYPE_STRUCT_SOCKADDR	35 /* struct sockaddr */
-#define PARAMETER_TYPE_FD_SET	36 /* fd_set: char **set, int *nfds, fd_set **in, fd_set **out */
-#define PARAMETER_TYPE_STRUCT_HOSTEN	37 /* struct hosten */
-#define PARAMETER_TYPE_IP_ADDR	38 /* ip addr: void **addr, int *type */
-#define PARAMETER_TYPE_STRUCT_ADDRINFO	39 /* struct addrinfo */
+	PARAMETER_TYPE_PERM,				/* write permission  */
+	PARAMETER_TYPE_STRUCT_STAT,			/* struct stat */
+	PARAMETER_TYPE_STRUCT_SOCKADDR,		/* struct sockaddr */
+	PARAMETER_TYPE_FD_SET,				/* fd_set: char **set, int *nfds, fd_set **in, fd_set **out */
+	PARAMETER_TYPE_STRUCT_HOSTEN,		/* struct hosten */
+	PARAMETER_TYPE_IP_ADDR,				/* ip addr: void **addr, int *type */
+	PARAMETER_TYPE_STRUCT_ADDRINFO,		/* struct addrinfo */
+	PARAMETER_TYPE_SOCKET_DOMAIN,		/* socket domain */
+	PARAMETER_TYPE_SOCKET_TYPE,			/* socket types */
+	PARAMETER_TYPE_SOCKET_OPTION,		/* socket option */
+	PARAMETER_TYPE_PROTO_LEVEL
+};
```

Comment:
Personally I preferred the `#defines` in this case because of the bitwise operations with `PARAMETER_FLAG_`s and confusing `enum` values that aren't explicitly listed in the declaration.

Reply:
@drystone I didn't checkout that case, will revert to `#define`